src/views/SignTransactionLedger.vue

```diff
@@ -175,17 +191,34 @@ export default class SignTransactionLedger extends Vue {
     private async mounted() {
         const network = this.$refs.network as Network;
 
-        let validityStartHeight;
+        let recipient: Nimiq.Address;
+        let value: number;
+        let fee: number;
+        let flags: number;
+        let validityStartHeight: number;
+
         if (this.request.kind === RequestType.SIGN_TRANSACTION) {
             const signTransactionRequest = this.request as ParsedSignTransactionRequest;
+            recipient = signTransactionRequest.recipient;
+            value = signTransactionRequest.value;
+            fee = signTransactionRequest.fee;
+            flags = signTransactionRequest.flags;
             validityStartHeight = signTransactionRequest.validityStartHeight;
         } else if (this.request.kind === RequestType.CHECKOUT) {
             const checkoutRequest = this.request as ParsedCheckoutRequest;
+            // Once currency selection is implemented likely no longer needed
+            const nimiqPaymentOption = (checkoutRequest.paymentOptions.find(
+                    (option) => option.currency === Currency.NIM,
+                ) as ParsedNimiqDirectPaymentOptions);
+            recipient = nimiqPaymentOption.protocolSpecific.recipient!;
+            value = nimiqPaymentOption.amount;
+            fee = nimiqPaymentOption.protocolSpecific.fee!;
+            flags = nimiqPaymentOption.protocolSpecific.flags;
```

Comment:
Maybe it works here, too? (I have not tried this, just feel this can work.)

```ts
{ recipient!, fee!, flags } = nimiqPaymentOption.protocolSpecific;
value = nimiqPaymentOption.amount;
```